ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/ResourceMapper.java

```diff
@@ -0,0 +1,263 @@
+import io.ballerina.openapi.service.mapper.diagnostic.OpenAPIMapperDiagnostic;
+import io.ballerina.openapi.service.mapper.model.AdditionalData;
+import io.ballerina.openapi.service.mapper.model.OperationBuilder;
+import io.ballerina.openapi.service.mapper.parameter.ParameterMapper;
+import io.ballerina.openapi.service.mapper.parameter.ParameterMapperInterface;
+import io.ballerina.openapi.service.mapper.response.ResponseMapper;
+import io.ballerina.openapi.service.mapper.response.ResponseMapperInterface;
+import io.ballerina.openapi.service.mapper.utils.MapperCommonUtils;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.Operation;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+
+import static io.ballerina.openapi.service.mapper.Constants.DEFAULT;
+import static io.ballerina.openapi.service.mapper.utils.MapperCommonUtils.getOperationId;
+
+/**
+ * This class will do resource mapping from ballerina to openApi.
+ *
+ * @since 2.0.0
+ */
+public class ResourceMapper implements ResourceMapperInterface {
+    private final Paths pathObject = new Paths();
+    private final AdditionalData additionalData;
+    private final OpenAPI openAPI;
+    private final List<FunctionDefinitionNode> resources;
+    private final boolean treatNilableAsOptional;
+
+    /**
+     * Initializes a resource parser for openApi.
+     */
+    ResourceMapper(OpenAPI openAPI, List<FunctionDefinitionNode> resources, AdditionalData additionalData,
+                   boolean treatNilableAsOptional) {
+        this.openAPI = openAPI;
+        this.resources = resources;
+        this.additionalData = additionalData;
+        this.treatNilableAsOptional = treatNilableAsOptional;
+    }
+
+    public void addMapping() {
+        Components components = openAPI.getComponents();
+        if (components == null) {
+            components = new Components();
+            openAPI.setComponents(components);
+        }
+        if (components.getSchemas() == null) {
+            components.setSchemas(new TreeMap<>());
+        }
+        for (FunctionDefinitionNode resource : resources) {
+            generateResourceMapping(resource, components);
+        }
+        if (components.getSchemas().isEmpty()) {
+            openAPI.setComponents(null);
+        }
+        openAPI.setPaths(pathObject);
+    }
+
+    private void generateResourceMapping(FunctionDefinitionNode resource, Components components) {
+        String path = MapperCommonUtils.unescapeIdentifier(generateRelativePath(resource));
+        String httpMethod = resource.functionName().toString().trim();
+        if (httpMethod.equals("'" + DEFAULT) || httpMethod.equals(DEFAULT)) {
```

Comment:
```suggestion
        if (httpMethod.equals(String.format("'%s", DEFAULT)) || httpMethod.equals(DEFAULT)) {
```

Reply:
Is there a advantage for using `String.format` rather than using a simple string concatenation?

Reply:
No advantage here, but it looks cleaner than the string concatenation.

Reply:
Fixed via [de24c2d](https://github.com/ballerina-platform/openapi-tools/pull/1564/commits/de24c2d24266204eb9f5179409534ecf373f45a4)